middle_end/flambda2/simplify/expr_builder.ml

```diff
@@ -520,18 +520,36 @@ let create_let_symbols uacc lifted_constant ~body =
         | None ->
           let prim : P.t =
             let symbol = Simple.symbol (Symbol_projection.symbol proj) in
+            let kind = Symbol_projection.kind proj in
             match Symbol_projection.projection proj with
             | Block_load { index } ->
               let index = Simple.const_int index in
               let block_access_kind : P.Block_access_kind.t =
-                Values
-                  { tag = Known Tag.Scannable.zero;
-                    size = Unknown;
-                    field_kind = Any_value
-                  }
+                match Flambda_kind.With_subkind.kind kind with
+                | Value ->
+                  let field_kind : P.Block_access_field_kind.t =
+                    if Flambda_kind.With_subkind.equal kind
+                         Flambda_kind.With_subkind.tagged_immediate
+                    then Immediate
+                    else Any_value
+                  in
+                  Values
+                    { tag = Known Tag.Scannable.zero;
```

Comment:
This is not directly linked to this PR, but I don't understand why `Known Tag.Scannable.zero` is correct here.

Reply:
Fixed.  I think `Unknown` here is ok (`To_cmm` ignores this in particular, I think).